tests/unit/LSYiiValidatorsTest.php

```diff
@@ -27,4 +27,155 @@ public function testHtmlEntityDecodeFilter()
+
+        //Returning to original values.
+        \Yii::app()->setConfig('filterxsshtml', true);
+        \Yii::app()->session['loginID'] = null;
+
+        //Delete user.
+        $user = \User::model()->findByPk($userId);
+        $user->delete();
+    }
+
+    /**
+     * Testing that any script or dangerous HTML is removed.
+     */
+    public function testXssFilterApplied()
+    {
+        $validator = new \LSYii_Validators();
+
+        $cases = array(
+            array(
+                'string'   => '<script>alert(`Test`)</script>',
+                'expected' => ''
+            ),
+            array(
+                'string'   => `{join('html_entity_decode("', '<script>alert("Test")</script>")')}`,
+                'expected' => ''
+            ),
+            array(
+                'string'   => '<title>html_entity_decode("<script>alert("Test")</script>")</title>',
+                'expected' => 'html_entity_decode("")'
+            ),
+            array(
+                'string'   => `{join('html_entity_decode("', '<s', 'cript>alert("Test")</script>")')}`,
+                'expected' => ''
+            ),
+            array(
+                'string'   => `{join('html_entity_decode("', '<', 'script>alert("Test")<', '/script>")')`,
+                'expected' => ''
+            ),
+            array(
+                'string'   => '<title>html_entity_decode("<script>alert("Test")</script>123456")</title>',
+                'expected' => 'html_entity_decode("123456")'
+            )
+        );
+
+        foreach ($cases as $case) {
+            $this->assertSame($case['expected'], $validator->xssFilter($case['string']), 'Unexpected filtered dangerous string.');
+        }
+    }
+
+    /**
+     * Testing that safe HTML tags are not removed.
+     */
+    public function testSafeHtml()
+    {
+        $validator = new \LSYii_Validators();
+
+        $cases = array(
+            '<p>Paragraph</p>',
+            '<strong>Text</strong>',
+            '<span>Some text</span>',
+        );
+
+        foreach ($cases as $case) {
+            $this->assertSame($case, $validator->xssFilter($case), 'Unexpected filtered safe HTML tags.');
+        }
+    }
+
+    /**
+     * Testing the language filters
+     * through the Survey model.
+     */
+    public function testLanguageFilters()
+    {
+        // Testing languageFilter.
+        $survey = \Survey::model()->insertNewSurvey(array('language' => 'ko')); // Set language to Korean.
+
+        $this->assertSame('ko', $survey->language, 'The language filter did not return a correctly filtered language string.');
+
+        $survey->language = 'enǵ';
+        $survey->save();
+
+        $this->assertSame('en', $survey->language, 'The language filter did not return a correctly filtered language string.');
+
+        // Testing multiLanguageFilter.
+        $survey->additional_languages = 'es';
```

Comment:
Why not `en es` here ? Must work

Reply:
`en` is the main language.
There is now an assert for more than 2 languages.